### DISCUSSION TITLE
Is there a method for binding/postprocessing custom Functions, rather than custom Types?

### DISCUSSION THREAD

**Discussion by doulighan** [sqlalchemy/sqlalchemy#11596] [Usage Questions] [answered]
I use TypeDecorator, process_bind_params and process_result_value to declare custom types. This works great, but I'm now in a situation where I would like something similar for the return value of a Function. 

For example in MySQL, the result of the function `GREATEST(value, NULL)` is `NULL`. In other specs, it would be `value`. You can (hackily) avoid this by running:
``` 
/// pseudocode
MIN_INT = -2,147,483,648
result = execute( 'SELECT GREATEST(COALESCE(value, :MIN_INT), COALESCE(NULL, :MIN_INT)) ')

if result = MIN_INT
  return None
return result
```

It would be nice if I could somehow have a user-defined `func.GREATEST_COALLESCE()` which automatically does this binding and result checking. Of course postprocessing would only work on top-level SELECTs, I don't think we would want to emit CASE() statements to make comparison DB side. Not exactly clear how this would work in edge cases.

Either way, is there an existing solution for this?

**Comment by zzzeek** [comment, marked as answer]
well yes, normally you bind your type to the function directly, `func.GREATEST_COALESCE(..., type_=MyType)`, however you can also register `GREATEST_COALESCE`  into the func namespace by subclassing [GenericFunction](https://docs.sqlalchemy.org/en/20/core/functions.html#sqlalchemy.sql.functions.GenericFunction).
